posthog-core/src/index.ts

```diff
@@ -97,6 +98,7 @@ export abstract class PostHogCoreStateless {
       retryCheck: isPostHogFetchError,
     }
     this.requestTimeout = options?.requestTimeout ?? 10000 // 10 seconds
+    this.featureFlagsRequestTimeout = options?.featureFlagsRequestTimeout ?? 3000 // 3 seconds
```

Comment:
3s might be very low for emerging countries, I know it's configurable but wondering if it's a good default, maybe 5s? thinking of Mobile specifically where the internet is much worse.

Reply:
More here https://github.com/PostHog/posthog-js-lite/pull/197#issuecomment-1972690919

Reply:
I'm basing the timeout on the request latency + external monitor latency: https://status.posthog.com/

But you're right the external monitor probably is also close by.

Although, node should be on a server, maybe we can have a different larger default for react-native? My worry is it might be too long on any server if it blocks for 5s by default.

Reply:
That's why some HTTP clients have more timeout configurations, such as connect (which involves DNS resolving, SSL handshaking, etc), read, write, etc.

Apparently, here the abort signal is all or nothing which is suboptimal.

Let's keep it 10s for RN, I already had issues with 5s default with Mobile SDKs, and 10s was the magic number that worked better for emerging countries or very slow/spotty connections.

Reply:
Reasonable, will keep the default for RN and let people customise it 👍